components/org.wso2.carbon.datasource.core/src/test/java/org/wso2/carbon/datasource/core/BaseTest.java

```diff
@@ -25,26 +35,44 @@
  */
 public class BaseTest {
     protected DataSourceManager dataSourceManager;
+    private static Logger logger = LoggerFactory.getLogger(BaseTest.class.getName());
 
     protected void init() throws DataSourceException {
         setEnv();
-        Path configPathCopyLocation = Paths.get("target", "carbonHome", "conf", "datasources");
+/*        Path configPathCopyLocation = Paths.get("target", "carbonHome", "conf", "datasources");*/
+        clearDeploymentConfiguration();
+        ConfigProvider configProvider = new ConfigProviderImpl(new YAMLBasedConfigFileReader("deployment.yaml"));
         dataSourceManager = DataSourceManager.getInstance();
-        dataSourceManager.initDataSources(configPathCopyLocation.toFile().getAbsolutePath());
+        dataSourceManager.initDataSources(configProvider);
     }
 
     private void setEnv() {
         Path carbonHomePath = Paths.get("target", "carbonHome");
         System.setProperty("carbon.home", carbonHomePath.toFile().getAbsolutePath());
 
-        Path configFilePath = Paths.get("src", "test", "resources", "conf", "datasources", "master-datasources.xml");
-        Path configPathCopyLocation = Paths.get("target", "carbonHome", "conf", "datasources",
-                "master-datasources.xml");
+        Path configFilePath = Paths.get("src", "test", "resources", "conf", "deployment.yaml");
```

Comment:
Should we have deployment.yaml file name as a constant in this class (Since the same file name is used in multiple places)